Userland/Libraries/LibAudio/Loader.h

```diff
@@ -103,7 +103,7 @@ class Loader : public RefCounted<Loader> {
         return m_plugin->seek(position);
     }
 
-    int loaded_samples() const { return m_plugin->loaded_samples(); }
+    int loaded_samples() const { return m_plugin->loaded_samples() - (int)m_buffer.size(); }
```

Comment:
good catch, I'm pretty sure that bugged out aplay a little. And yes, get_more_samples must only regard the plugin's state since it makes a decision on whether it can into the plugin's loader functionality.

Reply:
Are you saying that `Loader::get_more_samples()` needs to be changed? Doesn't the variable `remaining_samples` need to include the `Loader::m_buffer.size()` since it is used to determine how many samples are pulled from said buffer?

(You did get me to look back at that function, though, and I realized it was early exiting if the loader had stopped providing buffers but there was still data in `m_buffer`)